### PR TITLE
mds: master/slave -> leader/peer

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by fullerdj** [comment]
I think we could be a little clearer here. In this context, we're referring to an MDS that needs more information to satisfy a request.

I'd suggest master/slave -> handler/peer, which more clearly reflects that the MDS is querying others for information.

File: src/mds/events/EHelperUpdate.h [hunk 1]
```diff
@@ -99,7 +99,7 @@ WRITE_CLASS_ENCODER(rename_rollback::drec)
 WRITE_CLASS_ENCODER(rename_rollback)
 
 
-class ESlaveUpdate : public LogEvent {
+class EHelperUpdate : public LogEvent {
```
Comment (batrick) [inline src/mds/events/EHelperUpdate.h:102]: I like ` EPeerUpdate` better, what do you think?

File: src/messages/MMDSHelperRequest.h [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
-#ifndef CEPH_MMDSSLAVEREQUEST_H
-#define CEPH_MMDSSLAVEREQUEST_H
+#ifndef CEPH_MMDSHELPERREQUEST_H
+#define CEPH_MMDSHELPERREQUEST_H
 
 #include "mds/mdstypes.h"
 #include "messages/MMDSOp.h"
 
-class MMDSSlaveRequest : public MMDSOp {
+class MMDSHelperRequest : public MMDSOp {
```
Comment (batrick) [inline src/messages/MMDSHelperRequest.h:22]: same suggestion: `MMDSPeerRequest` ?

File: src/mds/Mutation.h [hunk 3]
```diff
@@ -158,8 +158,8 @@ struct MutationImpl : public TrackedOp {
     return lock == last_locked;
   }
 
-  bool is_master() const { return slave_to_mds == MDS_RANK_NONE; }
```
Comment (batrick) [inline src/mds/Mutation.h:161]: In this context, would `is_authority` be clearer?

Comment (ukernel) [reply src/mds/Mutation.h:161]: I don't think so

**Comment by batrick** [comment]
@liewegas did you have an update to ceph-object-corpus?

**Comment by liewegas** [comment]
Oops!  Pushed that too.

**Comment by batrick** [comment]
jenkins retest this please

**Comment by batrick** [comment]
jenkins test dashboard backend

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200717.003135

Unrelated new failures:

- https://tracker.ceph.com/issues/46609
- https://tracker.ceph.com/issues/46608

**Pull request merged**